workers/unity/Packages/com.improbable.gdk.core/Logging/ForwardingDispatcher.cs

```diff
@@ -11,12 +11,31 @@ namespace Improbable.Gdk.Core
     /// </summary>
     public class ForwardingDispatcher : ILogDispatcher
     {
-        public Connection Connection { get; set; }
-
         private readonly LogLevel minimumLogLevel;
 
         private bool inHandleLog;
 
+        private bool handlingUnityLogMessages;
+        private Connection connection;
+
+        public Connection Connection
+        {
+            get => connection;
+            set
+            {
+                if (connection == null)
+                {
+                    handlingUnityLogMessages = true;
+                    Application.logMessageReceived += LogCallback;
+                }
+                else if (value == null)
+                {
+                    handlingUnityLogMessages = false;
```

Comment:
what is this bool needed for?

Reply:
I'm going to rewrite all of this though now that we've made it disposable. Previously I'd made the worker responsible for the lifecycle of the logger.